### PR TITLE
Add mod_aggregate to ssh_auth to manage authorized_keys

### PR DESCRIPTION
**Pull request by mchugh19**
Not sure if this is an appropriate way to handle this scenario. Comments requested!

### What does this PR do?
Adds a mod_aggregate function to the ssh_auth state. This then builds up a list of the ssh keys from this and subsequent ssh_auth.present declarations, for each user. The present function then uses this list to determine what ssh_keys are being managed by salt, and removes the rest.

### What issues does this PR fix or reference?
#13340

### Previous Behavior
ssh_auth.present was only able to add keys. Any old keys not managed by salt need to be removed by specifying them in a ssh_auth.absent state.

### New Behavior
By specifying `aggregate: True` for the ssh_auth.present state, ssh keys not added by salt will be removed.

####  Example
When adding the Asomekeyh ssh key, another Aoldkey= was found for this user. Since it was not provisioned by salt, it was removed.

Here is a state to ensure that Asomekeyh is present. In the past Aoldkey= had been added to the host, but has been forgotten and is not managed by salt.
```
key1:
  ssh_auth.present:
    - user: testuser
    - enc: ssh-rsa
    - comment: testuser 1
    - aggregate: True
    - names:
      - Asomekeyh
      - MotherkeyB

#key2:
#  ssh_auth.present:
#    - user: testuser
#    - enc: ssh-rsa
#    - comment: 2
#    - names:
#      - Aoldkey=
```

By running this state, ssh_auth.present ensured the Asomekeyh was present, but also removed any keys found for the testuser user which were not managed by salt.
```
          ID: key1
    Function: ssh_auth.present
        Name: Asomekeyh
      Result: True
     Comment: The authorized host key Asomekeyh is already present for user testuser
     Started: 16:08:57.859383
    Duration: 9.498 ms
     Changes:
              ----------
              Aoldkey=:
                  Key removed
```
### Tests written?
No

### Commits signed with GPG?
No

### CODE REVIEW

File: salt/states/ssh_auth.py
Comment (rallytime) [inline salt/states/ssh_auth.py]: Can you change this to `Neon`?

**Review by rallytime — COMMENTED** [review]
I have one small comment here.

File: salt/states/ssh_auth.py
Comment (gtmanfred) [inline salt/states/ssh_auth.py]: Can we expose this so that it can be used directly as well, instead of only through mod_aggregate?

**Review by gtmanfred — CHANGES_REQUESTED** [review]

File: salt/states/ssh_auth.py
Comment (mchugh19) [reply salt/states/ssh_auth.py]: Whoops! Yep, will do.

File: salt/states/ssh_auth.py
Comment (gtmanfred) [reply salt/states/ssh_auth.py]: Ok, removing the ssh keys is not the purpose of mod_aggregate.

The purpose of mod_aggregate is to combine all of the calls into just one call.

See how it is used in pkg.installed where it makes only one call to the system package manager, instead of once per package state.

I am not sure that we should make this change if it is going to start removing keys that are not managed by the aggregate.  @saltstack/team-core opinions?

Comment (mchugh19) [reply salt/states/ssh_auth.py]: Agreed. Hence the initial "I don't know if this is kosher" comment. 

In order to perform authorized_keys management where keys not managed by salt are removed, we would need to know about all the keys that salt is managing. Today that means a series of `ssh_auth.present` stanzas. Because they are indepent, it forces the use of mod_aggregate. This overloads the use of mod_aggregate. In the pkg state it aggregates calls while in this use it aggregates and allows the multiple calls to more completely manage the authorized_keys file. Possibly a hack, or possibly awesome?!

If there are cleaner ways of accomplishing this, I'm for that as well. But a definitely a review is needed before any potential merge.

Comment (gtmanfred) [reply salt/states/ssh_auth.py]: yeah, i just want to have more input from the rest of the core team to see if we are ok with this.

**Review by terminalmage — COMMENTED** [review]
This is _definitely_ not how we want to use aggregate.

**Comment by mchugh19** [comment]
@gtmanfred  @terminalmage @cachedout is there a recommendation for how to support #13340 functionality?

`ssh_auth.present` allows for passing various arguments and options, which is why this PR only attempted to handle removing unlisted keys and continued to use independent ssh_auth.present declarations to add. Perhaps for the simple use-case an "ssh-keys" key could be created which adds those keys, and removes all others? This would require all keys listed to have the same options, but perhaps that is an acceptable trade off?

**Comment by mchugh19** [comment]
Closing this out in favor of a different approach

**Pull request closed without merging**
